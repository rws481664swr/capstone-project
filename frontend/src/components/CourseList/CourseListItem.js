import css from './index.css'
import {Link} from "react-router-dom";

const CourseListItem = ({course: {courseName, subject, _id, courseNumber, students,...rest}}) => {

    return <>
        <li className={`${css.courseLi}`}>
            <Link to={`/courses/${_id}`}>
                <div className={css.courseBody}>
                    <div className={css.courseName}>{subject}:#{courseNumber} - {courseName} </div>
                    <div className={css.courseName}>{students.length} Students</div>
                </div>
                <h1>{_id}</h1>
            </Link>
        </li>

    </>
}
export default CourseListItem

