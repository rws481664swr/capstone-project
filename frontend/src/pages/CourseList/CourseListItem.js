import {Link} from "react-router-dom";
import useDropCourse from "../../hooks/ajax/useDropCourse";
import {DeleteButton} from "../../components/General/Button/IconButtons/IconButtons";

const CourseListItem = ({drop,course: {courseName, subject, _id, courseNumber, students, ...rest}}) => {

    const dropCourse= drop(useDropCourse(_id))
    return(

        <li className={`CourseListItem`}>
                <Link to={`/courses/${_id}`}>
                    <div className={'CourseListItem_Body'}>
                        <div>
                            <div>{subject}:#{courseNumber} - {courseName} </div>
                            <div>{students.length} Students</div>
                        </div>
                        <div>
                             <DeleteButton onClick={dropCourse}/>
                        </div>
                    </div>
                </Link>
        </li>
)


}
export default CourseListItem

