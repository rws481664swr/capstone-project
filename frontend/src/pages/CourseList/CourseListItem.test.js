import {render} from "@testing-library/react";
import CourseListItem from "./CourseListItem";
import {MemoryRouter} from "react-router-dom";

it('should render CourseListItem', () => {
    const {getByText} = render(
       <MemoryRouter>
        <CourseListItem
            course={
            {courseName: 'testName',
                subject: 'testSubj',
                _id: 'test',
                courseNumber: 'test',
                students: [],
                teachers:[]
            }}/>
       </MemoryRouter>)
    expect(getByText('0 Students')).toBeInTheDocument()
    expect(getByText(`testSubj:#test - testName`)).toBeInTheDocument()
})
it('should render CourseListItem', () => {
    const {asFragment} = render(
       <MemoryRouter>
        <CourseListItem
            course={
            {courseName: 'testName',
                subject: 'testSubj',
                _id: 'test',
                courseNumber: 'test',
                students: [],
                teachers:[]
            }}/>
       </MemoryRouter>)
    expect(asFragment()).toMatchSnapshot()
})
