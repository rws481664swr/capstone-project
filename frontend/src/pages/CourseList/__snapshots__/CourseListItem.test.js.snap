// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CourseListItem 1`] = `
<DocumentFragment>
  <li
    class="undefined"
  >
    <a
      href="/courses/test"
    >
      <div>
        <div>
          testSubj:#test - testName 
        </div>
        <div>
          0 Students
        </div>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
